Keep DropdownMenu open after selection

riko111 opened the pull request:
## Issue
- close #598
- close #500 

## Overview (Required)
- Keep DropdownMenu open after item selection to allow multiple selections

## Links
- 

## Movie (Optional)
Before | After
:--: | :--:
<video src="https://github.com/user-attachments/assets/4a82404b-a1ba-49ec-84ce-2f5c47b197f7" width="300" > | <video src="https://github.com/user-attachments/assets/64af93d5-8e0c-43f0-99a5-c416a774278d" width="300" >

riko111 committed:
Keep DropdownMenu open after item selection to allow multiple selections

kitakkun reviewed (approved):
Looks great! Thank you. 👍🏻

riko111 committed:
Merge branch 'DroidKaigi:main' into main

riko111 committed:
Merge branch 'DroidKaigi:main' into main

riko111 committed:
Merge branch 'DroidKaigi:main' into main

kitakkun commented:
@riko111
Thanks so much for your work on this! After discussing with our team, we think it might be best if this feature is implemented with the following behavior:

- When a dropdown menu has multiple options, long-pressing a specific item could enter multi-selection mode.  
  - We suggest adding Haptic feedback to indicate when entering that mode.
- Once in multi-selection mode, selecting other items shouldn’t close the dropdown menu.

Would you be able to update your implementation to match these specifications? Thanks in advance!

riko111 committed:
Merge branch 'DroidKaigi:main' into main

riko111 committed:
Add long-press multi-selection mode for category items in DropdownMenu

riko111 committed:
Apply spotless formatting

kitakkun commented on feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/components/SearchFilterRow.kt:
```diff
@@ -176,25 +205,40 @@ private fun <T> FilterDropdown(
+                        ),
+                    verticalAlignment = Alignment.CenterVertically,
+                ) {
+                    if (item in selectedItems) {
+                        Icon(Icons.Default.Check, contentDescription = null, modifier = Modifier.padding(end = 12.dp))
+                    }
```
Hiding the icon entirely when unselected might cause the view width to shift compared to the existing behavior, which could feel janky. Maybe controlling the alpha instead would be smoother.

https://github.com/user-attachments/assets/d3f0d5f4-3db9-43c5-a706-a203aec4860f

kitakkun commented:
> By the way, do we still need the session type dropdown list? It might be unnecessary, especially since it doesn't exist in the iPhone version.

There’s a corresponding issue opened for this: #500.

riko111 commented on feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/components/SearchFilterRow.kt:
```diff
@@ -50,6 +54,7 @@ const val DropdownFilterChipTestTagPrefix = "DropdownFilterChipTestTag:"
 fun SearchFilterRow(
     filters: SearchScreenUiState.Filters,
     onFilterToggle: (SearchScreenEvent.Filter) -> Unit,
+    onFilterLongPress: ((SearchScreenEvent.Filter) -> Unit)? = null,
```
Thanks for pointing that out, I hadn’t noticed it.  
I’ll remove the parameter along with the comment below.

riko111 committed:
Remove unused onFilterLongPress and simplify multi-selection behavior

riko111 committed:
Merge branch 'DroidKaigi:main' into main

riko111 committed:
Remove unused onFilterLongPress and simplify multi-selection behavior

riko111 committed:
Keep layout stable by fading out unselected icons and locking chip width while menu is open

- Fade out unselected check icons instead of removing them to avoid width shifts
- Lock the chip width while the dropdown menu is expanded so the anchor position stays stable

riko111 committed:
Keep layout stable by fading out unselected icons and locking chip width while menu is open

- Fade out unselected check icons instead of removing them to avoid width shifts
- Lock the chip width while the dropdown menu is expanded so the anchor position stays stable

riko111 committed:
Remove sessionType dropdown list from tests after component removal

riko111 commented:
Addressed the review feedback and updated the implementation accordingly.

https://github.com/user-attachments/assets/907611b0-6f78-4966-a0ed-e0f7fb6dea75

kitakkun commented on feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/components/SearchFilterRow.kt:
```diff
@@ -126,11 +119,17 @@ private fun <T> FilterDropdown(
     items: List<T>,
     itemLabel: (T) -> String,
     onItemSelected: (T) -> Unit,
+    onMultiSelectFinished: (List<T>) -> Unit = {},
```
Specifying default values for lambda parameters has some downsides: it can make it easy to forget to provide a value, and at a glance it’s not always clear whether a value is actually being passed. Therefore, unless there’s a specific reason, I’d prefer to remove default empty lambdas.

riko111 committed:
Anchor DropdownMenu to a stable empty Box (menuAnchor) and remove width-lock logic

riko111 committed:
Remove default empty lambda; make onMultiSelectFinished nullable and invoke only when provided

riko111 committed:
Revert "Remove default empty lambda; make onMultiSelectFinished nullable and invoke only when provided"

This reverts commit 3c3894b90ea0b7ec44271d87cb40416b1c123086.

riko111 committed:
Merge branch 'DroidKaigi:main' into main

riko111 committed:
spotless-check

riko111 committed:
Merge remote-tracking branch 'origin/main'

kitakkun commented on feature/sessions/src/commonMain/kotlin/io/github/droidkaigi/confsched/sessions/components/SearchFilterRow.kt:
```diff
@@ -176,25 +188,44 @@ private fun <T> FilterDropdown(
+                            .alpha(if (selectedItems.contains(item)) 1f else 0f),
+                    )
+                    Text(itemLabel(item), style = MaterialTheme.typography.bodyLarge)
+                }
```
I’m not sure why, but it seems that adding the following Modifier to the DropdownMenuItem not to the outer Box allows the existing tests to pass without issues:

```kotlin
modifier = Modifier.testTag(DropdownFilterChipTestTagPrefix.plus(item))
```

kitakkun commented:
@riko111 
Since I got a review request, I just wanted to ask—maybe there’s a way to resolve the test failure with the approach I suggested in [my previous comment](https://github.com/DroidKaigi/conference-app-2025/pull/604#discussion_r2337084749)?

kitakkun commented:
Just a quick comment.

If it’s only the changes for Issue #500, it could be merged independently, so feel free to consider doing that!

riko111 commented:
Sorry, I think I mistakenly sent a review request. I haven’t had a chance to look into it yet, so please withdraw it.
